Add copy about uninstalling the application

icancclearynow opened the pull request:
I've attempted to add some copy about uninstalling the application to address issue https://github.com/elementary/docs/issues/97.

I've compiled what I believe uninstalls the application, but I am following the tutorial while proposing these edits.  So if one of the maintainers would advise/edit as needed, I'd be happy to update my branch.

-----

### Uninstalling the application

To uninstall your application, we need to remove the several files we've created outside of our project folder:

* Binary file located at /usr/bin/com.github.yourusername.hello-again
* Desktop file in /usr/share/applications/com.github.yourusername.hello-again.desktop
* Application Data file in /usr/share/metainfo/com.github.yourusername.hello-again.appdata.xml

```bash
sudo rm /usr/bin/com.github.yourusername.hello-again
sudo rm /usr/share/applications/com.github.yourusername.hello-again.desktop
sudo rm /usr/share/metainfo/com.github.yourusername.hello-again.appdata.xml
```

Eldhrimer commented:
I'm not sure about Application Data, but doesn't ```sudo ninja uninstall``` inside the build folder removes the necessary files?

icancclearynow commented:
It wasn't clear to me how to uninstall. I'd like the docs to state the method directly.

If `sudo ninja uninstall` from the build folder is the right way, I'd be happy to change the PR contents.

Is that the recommended way?

Thanks!

icancclearynow commented:
@Eldhrimer `sudo ninja uninstall` from the `build` folder worked.  I agree that is the preferred way forward.

```bash
sudo ninja uninstall
[0/1] /usr/bin/meson --internal uninstall
Deleted: /usr/bin/com.github.icancclearynow.nordvpn
Deleted: /usr/share/applications/com.github.icancclearynow.nordvpn.desktop
Deleted: /usr/share/metainfo/com.github.icancclearynow.nordvpn.appdata.xml
Deleted: /usr/share/icons/hicolor/16x16/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/16x16@2/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/24x24/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/24x24@2/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/32x32/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/32x32@2/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/48x48/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/48x48@2/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/64x64/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/64x64@2/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/128x128/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/icons/hicolor/128x128@2/apps/com.github.icancclearynow.nordvpn.svg
Deleted: /usr/share/glib-2.0/schemas/com.github.icancclearynow.nordvpn.gschema.xml

Uninstall finished.

Deleted: 16
Failed: 0

Remember that files created by custom scripts have not been removed.
```

I'll update the PR.

icancclearynow commented:
Per https://github.com/elementary/docs/pull/134/commits/4a747979560a586c780f499b64bb710e610db5c2, I added copy regarding the `flatpak remove` command.

Per https://github.com/elementary/docs/pull/134/commits/58a969a430a8f4ba8d6ec6f2e2b90844e92c1898, I added a link to Flatpak's documentation.

Eldhrimer commented on writing-apps/our-first-app/the-build-system.md:
````diff
@@ -69,6 +69,25 @@ If all went well, you should now be able to open your app from the Applications
+If all went well, you should see command output that shows files related to your application were removed.  Again, more details can be found in [Meson's documentation](https://mesonbuild.com/Manual.html).
+
+```bash
+...
````
The manual removal commands still appear in the copy, and the code block is without any context or explanation. I think that if you wanna keep the manual instructions for any reason at least provide context about what those commands do.

Eldhrimer commented on writing-apps/our-first-app/the-build-system.md:
```diff
@@ -69,6 +69,25 @@ If all went well, you should now be able to open your app from the Applications
+sudo rm /usr/bin/com.github.yourusername.hello-again
+sudo rm /usr/share/applications/com.github.yourusername.hello-again.desktop
+sudo rm /usr/share/metainfo/com.github.yourusername.hello-again.appdata.xml
+...
```
The manual removal commands still appear in the copy, and the code block is without any context or explanation. I think that if you wanna keep the manual instructions for any reason at least provide context about what those commands do.

icancclearynow replied:
Well, actually I was trying to give the user an indication of what they should see as output from the command, but I see how that is confusing and inconsistent. Most other bash code blocks are for the user to enter and run.

icancclearynow replied:
This should be resolved by removing the bash output in https://github.com/elementary/docs/pull/134/commits/8ff32d85c43b45423fc92bb3ce95b455653d977a

cassidyjames commented on writing-apps/our-first-app/the-build-system.md:
```diff
@@ -59,8 +59,8 @@ Change into the build directory and use `ninja` to build. Then, if the build is
 cd build
-ninja
-ninja install
+sudo ninja
+sudo ninja install
```
You definitely don't need to build as root, and `ninja install` will automatically use polkit to handle elevating privileges.

cassidyjames commented on writing-apps/our-first-app/the-build-system.md:
````diff
@@ -71,16 +71,21 @@ If you were about to add the "build" folder to your git repository and push it,
 sudo rm /usr/share/applications/com.github.yourusername.hello-again.desktop
 sudo rm /usr/share/metainfo/com.github.yourusername.hello-again.appdata.xml
+...
 ```
````
Something's gone awry here and you have double backtick blocks

cassidyjames commented on writing-apps/our-first-app/the-build-system.md:
````diff
@@ -71,16 +71,21 @@ If you were about to add the "build" folder to your git repository and push it,
-* Application Data file in `/usr/share/metainfo/com.github.yourusername.hello-again.appdata.xml`
+```bash
+cd build
+sudo ninja uninstall
````
I'm not sure, but I _think_ you can just `ninja uninstall` here as well, and it will elevate with polkit? I'd need to double-check, though.

Marukesu replied:
No, `ninja uninstall` do not use polkit, so sudo is needed.

Marukesu commented on writing-apps/our-first-app/the-build-system.md:
````diff
@@ -69,6 +69,17 @@ If all went well, you should now be able to open your app from the Applications
+To uninstall your application, change into the build directory and we will use `ninja` once again to uninstall:
+
+```bash
+cd build
+sudo ninja uninstall
````
You don't need to change to the build directory if using `ninja -C`:


```suggestion
sudo ninja -C build uninstall
```

The same is valid for the build and install lines.

iancleary commented:
@cassidyjames @Marukesu Thanks for the review!  

I believe I've incorporate your suggestions or requested changes.  Let me know if there is anything else you need.

Antolius commented on writing-apps/our-first-app/packaging.md:
````diff
@@ -62,3 +62,29 @@ This tells Flatpak Builder to build the manifest we just wrote into a clean `bui
+Flatpak will prompt you to remove your application.
+
+```bash
+flatpak remove com.github.yourusername.yourrepositoryname
````
I suggest going with `uninstall` command here, for a couple of reasons:

1. The `remove` option [is just an alias](https://github.com/flatpak/flatpak/blob/main/app/flatpak-main.c#L82-L83) for the `uninstall` command.
1. [Flatpak documentation](https://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpak-uninstall) only mentions the uninstall command
1.  Ninja also uses the uninstall terminology, so it would be consistent.

Plus, I suggest adding the `--delete-data` flag to the command, as leftover app data is unlikely to be useful, and might lead to complications in development.

These 2 changes would turn this line into:

```suggestion
flatpak uninstall com.github.yourusername.yourrepositoryname --delete-data
```

mrivasperez reviewed (commented):
This was really helpful for me. I hope your work is merged.

danirabbit reviewed (approved):
Sorry it took so long to get this merged. Nice work :)

iancleary commented:
> Sorry it took so long to get this merged. Nice work :)

You're welcome! Thanks for the merge 😊
